internal/postrs/log.go

```diff
@@ -0,0 +1,33 @@
+package postrs
+
+import "C"
+
+import (
+	"github.com/spacemeshos/post/shared"
+)
+
+var log shared.Logger
+
+//export logCallback
+type logCallback func(level C.int, message *C.char)
```

Comment:
This is just a proof of work for a sugared logger. A real structured logger is a bit more complex, we would require more than just one callback function. A simple `C` structured logger we can take expirations from is https://github.com/briandowns/struCtured-logger

@poszu do you want to log data from Rust or just messages? If it's just messages a simple interface like this is probably sufficient. If you want to log real data, that gets serialized to a proper json we should sync on how the interface should look like 🙂

Reply:
I was thinking about a simpler approach for now. Doing something more sophisticated would be overkill IMHO. Perhaps in the future? I was thinking about something like this: the callback would get a pointer to a structure containing some metadata (level, file, line, etc., and message string (already formatted)).

```go
// actually defined in C headers
type LogRecord struct {
  Level int
  Line int
  File String
  Message String
}

type logCallback func(record *LogRecord)
```

Reply:
That is fine by me too, depending on what you mean with formatted string. With structured logging I meant giving `postrs` the ability to log data not just a position and a string:

```
{"level": "debug", "file": "init.rs", "line": 1234, "msg": "found new vrf nonce", "commitment": "483d1413ae0cafd2b881fbedff00ac8ecb9e58dc193b7a2daef135c9f620380f", "nonce": 9876, "label_value": "00009d3292ae3d6097747f5d220d736da0fa30844e3f931e22f56747b10305c8"}
```

vs.

```
{"level": "debug", "file": "init.rs", "line": 1234, "msg": "found new vrf nonce; commitment: 483d1413ae0cafd2b881fbedff00ac8ecb9e58dc193b7a2daef135c9f620380f, nonce: 9876"}
```